src/include/storage/data_table.h

```diff
@@ -236,11 +240,19 @@ class DataTable {
   // The block compactor elides transactional protection in the gather/compression phase and
   // needs raw access to the underlying table.
   friend class BlockCompactor;
+  // The SqlTable needs to access the layout version
+  friend class SqlTable;
 
   const common::ManagedPointer<BlockStore> block_store_;
   const layout_version_t layout_version_;
   const TupleAccessStrategy accessor_;
 
+  // TODO(Schema-Change)
+  // Singly-linked list for tables with different version?
```

Comment:
Is this something that has yet to be decided? This seems a bit odd to do both

Reply:
right, we are going with the sqltable pointer approach now. good catch